src/Morpho.sol

```diff
@@ -377,10 +377,16 @@ contract Morpho is IMorpho {
         uint256 badDebtShares;
         if (position[id][borrower].collateral == 0) {
             badDebtShares = position[id][borrower].borrowShares;
-            uint256 badDebt = badDebtShares.toAssetsUp(market[id].totalBorrowAssets, market[id].totalBorrowShares);
-            market[id].totalSupplyAssets -= badDebt.toUint128();
-            market[id].totalBorrowAssets -= badDebt.toUint128();
-            market[id].totalBorrowShares -= badDebtShares.toUint128();
+            // Safe "unchecked" cast since the value is smaller than totalBorrowAssets.
+            uint128 badDebt = uint128(
+                UtilsLib.min(
+                    market[id].totalBorrowAssets,
+                    badDebtShares.toAssetsUp(market[id].totalBorrowAssets, market[id].totalBorrowShares)
+                )
+            );
```

Comment:
I'm more in favor of casting, because we do it everywhere already

Reply:
You mean safe casting?

Reply:
If yes, it's not really true. There are 4 other occurrences where we don't safe cast

Reply:
ah yes nice catch. I find it consistent to have this one "checked", but no veto

Reply:
Arbitrarily consistent then (I disagree for the same reason as Merlin)

Reply:
ok then let's just check everything and use only uint256 variables

Reply:
I assume you suggest this for consistency
What's the benefit of being consistent in this case?
I'd rather go for the status quo

Reply:
>I'd rather go for the status quo to minimize changes

agree but we disagree on what is the status quo here (in this PR to me we diverge from the status quo)

Reply:
Indeed, I meant I am against changing anything elsewhere (which I thought was what you meant with  "let's just check everything and use only uint256 variables")

And because we are modifying the line in this PR we have the free choice to either safe cast it or not
There seems to be no consistency in the rest of the code ; though we could argue that we TRIED to not safe cast when unnecessary - so I'd argue in favor of not safe casting it here

Reply:
@Rubilmax see #573 wdyt

Reply:
Let's merge this PR as it is and that's it, no?

Reply:
no let's just have a uint256 as it was, and safe cast

Reply:
🥲